### PR TITLE
Release: add release issue template to doc

### PR DESCRIPTION
**Pull request by pracucci**
#### What this PR does
The [RELEASE doc](https://github.com/grafana/mimir/blob/main/RELEASE.md) is great but also very verbose. I'm a big fan of checklists to follow instead, which I find quicker and easier. In this PR I'm proposing to add to the RELEASE doc the markdown of a checklist to copy-paste into a GitHub issue at the beginning of the release process. This is the checklist I've followed for 2.4.

#### Which issue(s) this PR fixes or relates to

N/A

#### Checklist

- [ ] Tests updated
- [x] Documentation added
- [ ] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

File: RELEASE.md
Comment (pstibrany) [inline RELEASE.md:68]: This step should be part of "create new release branch" imho.

File: RELEASE.md
Comment (pstibrany) [inline RELEASE.md:67]: This has nothing to do with "Publish the Mimir RC", and can be moved to earlier step. It should perhaps mention that this can be done on `main` branch, and final release note is later cherry-picked into the branch.

File: RELEASE.md
Comment (pstibrany) [inline RELEASE.md:87]: If this is part of public document, perhaps we can explain what "GEM" stands for for people who are not familiar with it.

File: RELEASE.md
Comment (pstibrany) [inline RELEASE.md:91]: Shall we merge release notes document into release branch too?

**Review by pstibrany — APPROVED** [review]
Thanks!

**Comment by pr00se** [comment]
What do you think of making releases a [new type of issue with its own template](https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/configuring-issue-templates-for-your-repository), rather than copying from the RELEASE document?

File: RELEASE.md
Comment (pracucci) [reply RELEASE.md:68]: I agre on the "main / unreleased" removal, but not sure about VERSION? VERSION needs to be updated for each release candidate. The idea is that you can copy-paste the "Publish the Mimir release candidate" sub checklist for each follow up release candidate.

**Comment by pracucci** [comment]
Thanks @pstibrany for the review! I pushed your suggestions in https://github.com/grafana/mimir/pull/3477/commits/378c7eae119a733238f1d91d562708ad97243f51. Could you take another look, please?

> What do you think of making releases a [new type of issue with its own template](https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/configuring-issue-templates-for-your-repository), rather than copying the checklist from the RELEASE document each time?

@pr00se That was my original plan, but I thought may be misleading for the community seeing we prompt the "release checklist" every time they open a new issue, so I proposed a simple copy-paste from the RELEASE doc.

**Comment by pr00se** [comment]
> @pr00se That was my original plan, but I thought may be misleading for the community seeing we prompt the "release checklist" every time they open a new issue, so I proposed a simple copy-paste from the RELEASE doc.

I understand the concern, but since we already prompt for different kinds of issues in the repo, I figured if we had something like "Mimir release" with a good description in wouldn't be that confusing. We could always try and remove it if it's confusing/annoying, but I defer to your judgement.

File: RELEASE.md
Comment (pstibrany) [reply RELEASE.md]: You're right, I removing meant "main / unreleased" part. VERSION is updated more often.

**Comment by pracucci** [comment]
> > @pr00se That was my original plan, but I thought may be misleading for the community seeing we prompt the "release checklist" every time they open a new issue, so I proposed a simple copy-paste from the RELEASE doc.
> 
> I understand the concern, but since we already prompt for different kinds of issues in the repo, I figured if we had something like "Mimir release" with a good description in wouldn't be that confusing. We could always try and remove it if it's confusing/annoying, but I defer to your judgement.

I don't know. We release once every 6 weeks. It's about copy-pasting a markdown into a release once every 6 weeks. I think the benefit of having a 1-click button instead of copy-pasting doesn't justify the misunderstanding we may create to OSS community that wants to open a PR about an issue with a Mimir release.

I don't feel strong about it, but I also wouldn't block on this discussion. It's something we can re-evaluate any time.

**Pull request merged**
